exploration-module/src/main/java/io/edpn/backend/exploration/infrastructure/kafka/KafkaTopicHandler.java

```diff
@@ -0,0 +1,44 @@
+package io.edpn.backend.exploration.infrastructure.kafka;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.ListTopicsResult;
+import org.apache.kafka.clients.admin.NewTopic;
+
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+
+@RequiredArgsConstructor
+public class KafkaTopicHandler {
+
+    private final AdminClient kafkaAdminClient;
+    private final int topicPartitions;
+    private final short topicReplicationFactor;
+
+
+    public CompletableFuture<String> createTopicIfNotExists(String topicName) {
```

Comment:
Since Kafka 1.1.0, `auto.create.topics.enable` has existed. Is there a reason we don't just let kafka/Ops manage topics?